manager/saw_manager.py

```diff
@@ -2,43 +2,154 @@
+    electric_saw2 = ElectricSaw(7.1, "R2-D2", 1700, 80, ("battery", "electricity"), True)
+    circular_saw2 = CircularSaw(6.6, "C3-PO", 1500, 1.05, 0.03, ("battery", "electricity"))
+    jigsaw2 = Jigsaw(8, "HALO-10000", 550, 0.45, 0.03)
+    list_of_saws = [chainsaw1, electric_saw1, circular_saw1, jigsaw1, chainsaw2,
+                    electric_saw2, circular_saw2, jigsaw2]
+    return list_of_saws
 
+
+def numb_of_args(func):
+    def inner(*args):
+        number_of_args = len(args)
+        print(f"number of args = {number_of_args}")
+        return func(*args)
+    return inner
+
+
+def numb_of_calls(func):
+    counter = 0
+
+    def inner(*args):
+        nonlocal counter
+        counter += 1
+        if counter >= 3:
+            raise Exception("Too many calls")
+        return func(*args)
+    return inner
+
+
+@numb_of_calls
+def convert_saws_to_str():
+    string_saws = []
+    for i in create_saws():
+        string_saws.append(i.__str__())
+    return string_saws
+
+
+@numb_of_calls
 def print_saws():
-    for i in list_of_saws:
+    """
+    printing all created saws
+    """
+    for i in create_saws():
         print(i)
 
 
+@numb_of_args
 def add_saw(saw):
-    list_of_saws.append(saw)
+    """
+    add a saw to list
+    """
+    create_saws().append(saw)
+
 
+@numb_of_args
+def find_saws_more_powerful_than(power):
+    """
+    :return: all saws, which power is more than param
+    """
+    list_of_powerful_saws = []
+    for i in create_saws():
+        if i.power > power:
+            list_of_powerful_saws.append(i)
+    return list_of_powerful_saws
 
+
+@numb_of_calls
 def find_all_working():
+    """
+    :return: all saws, which are working now
+    """
     list_of_working_saws = []
-    for i in list_of_saws:
+    for i in create_saws():
         if i.is_working is True:
             list_of_working_saws.append(i)
     return list_of_working_saws
 
 
-def find_saws_more_powerful_than(power):
-    list_of_powerful_saws = []
-    for i in list_of_saws:
-        if i.power > power:
-            list_of_powerful_saws.append(i)
-    return list_of_powerful_saws
+@numb_of_calls
+def print_len():
+    for i in create_saws():
+        print(i.brand.__len__())
+
+
+@numb_of_calls
+def print_getitem():
+    for i in create_saws():
+        print(i.__getitem__(i))
+
+
+@numb_of_calls
+def print_iter():
+    iter_saws = iter(create_saws())
+    for _ in create_saws():
+        print(next(iter_saws))
+
+
+@numb_of_calls
+def comprehension():
+    len_list = [i.brand.__len__() for i in create_saws()]
+    return len_list
+
+
+@numb_of_calls
+def enumerating():
+    for i in enumerate(convert_saws_to_str()):
+        print(i)
+
+
+@numb_of_calls
+def zipping():
+    zipped_list = zip(convert_saws_to_str(), comprehension())
+    print(list(zipped_list))
+
+
+@numb_of_calls
+def dictionary():
+    for i in create_saws():
+        dict_saw = i.__dict__
+        print(dict_saw)
+
+
+@numb_of_calls
+def all_or_any():
```

Comment:
this funcrion must use "all" and "any" built-in functions. Also consider giving variables better names